app/models/pod.rb

```diff
@@ -25,7 +25,9 @@ class Pod < ActiveRecord::Base
 
   class << self
     def find_or_create_by(opts) # Rename this method to not override an AR method
-      u = URI.parse(opts.fetch(:url))
+      url = opts.fetch(:url)
```

Comment:
I check this not because the key could be missing, but because it could present but be nil.
When called from [here](https://github.com/diaspora/diaspora/blob/develop/lib/hydra_wrapper.rb#L87) while running the testsuite response.effective_url in nil. That's why I had to add these lines.